### ISSUE TITLE
Embedded video cannot be maximised or played Picture-in-Picture

### ISSUE THREAD

**Issue by neil-fry** [SafeExamBrowser/seb-mac#226]
When using Safari, Chrome or Firefox on Mac, videos embedded within a webpage can be maximised. Safari also has the option to play Picture-in-Picture. However, when using seb-mac, these options are not available. This means that videos are sometimes too small to watch.

**Steps to reproduce**

1. Click on video - video starts to play
2. Observe that the no maximise or pip options are available within the video window.

Can't upload an example SEB file but the following can be used as an embedded video https://www.w3schools.com/html/html5_video.asp
Also attached is a screenshot of a video on safari

Not sure if this relates to an existing issue #178 

![Screenshot 2022-08-04 at 17 51 46](https://user-images.githubusercontent.com/87660904/182891686-2a0d1bd0-bdd8-4cdf-852c-c362d218fc99.png)

**Comment by danschlet** [comment]
Apple doesn't officially support full screen and picture-in-picture (PiP) video in WKWebView (the embedded browser which SEB is using). At least full screen should be possible to add [using private API](https://stackoverflow.com/questions/44988219/wkwebview-mac-browser-enable-fullscreen-html5), I'll have a look at it. Likely PiP might work similarly as well, but of course only inside of SEB as the Mac version currently isn't properly supporting open book exams running without single app mode along other apps.

**Comment by neil-fry** [comment]
Hi Dan, 
do you have any ideas if this may be implemented. I appreciate that you have lots of other issues. We are just trying to work out alternatives if this isn't happening in the near future. 
Cheers
Neil

**Comment by danschlet** [comment]
Hi Neil,

I already investigated it and [the bit](https://jira.let.ethz.ch/secure/RapidBoard.jspa?rapidView=73&view=detail&selectedIssue=SEBMAC-455&devStatusDetailDialog=repository) which is possible will be included in SEB 3.1.1, but it's not great: 

- Full screen is only possible when using the AAC Assessment Mode kiosk mode and the modern WebView, not with the regular SEB kiosk mode. AAC has had various bugs and it only works properly starting macOS 12.1 (there will be a workaround for macOS 11 and 10.15, requiring to use URL filters and define all domains which need to be accessed during the exam).
- Picture-in-Picture doesn't work with any kiosk mode. To make it work with AAC, Apple would need to fix it...

So in short it won't be possible to implement a perfect full screen solution inside SEB. I would recommend people to add some controls to their web applications to enlarge videos to the full size of the browser window.

**Comment by neil-fry** [comment]
Hi Dan,
thanks for letting me know.
Regards
Neil

**Comment by danschlet** [comment]
The solution described above is now available in [SEB 3.1.1rc](https://github.com/SafeExamBrowser/seb-mac/releases/tag/3.1.1rc) (release candidate, please test). As mentioned above, unfortunately there's nothing that can be done relating the limitations.
